src/bipartite.jl

```diff
@@ -0,0 +1,96 @@
+"""
+    BipartiteView{T<:Real} <: AbstractGraph{Int64}
+
+
+Creates a bipartite view of a hypergraph.
+Note this is a view - changes to the original hypergraph will be automatically reflected in the view.
+
+**Constructors**
+
+    BipartiteView{T}(h::Hypergraph{T}) where {T<:Real}
+
+Construct a bipartite view of a hypergraph for processing with LightGraphs
+
+
+**Arguments**
+
+* `h` : a hypergraph
+
+"""
+struct BipartiteView{T<:Real} <: AbstractGraph{Int64}
+    v2he::Vector{Dict{Int,T}}
+    he2v::Vector{Dict{Int,T}}
+    function BipartiteView{T}(h::Hypergraph{T}) where {T<:Real}
+        new(h.v2he,h.he2v)
+    end
+end
+
+
+LightGraphs.nv(b::BipartiteView) =
+     length(b.v2he)+length(b.he2v)
+
+LightGraphs.vertices(b::BipartiteView) =
+     Base.OneTo(nv(b))
+
+
+LightGraphs.ne(b::BipartiteView) =
+    sum(length.(b.v2he))+sum(length.(b.he2v))
```

Comment:
I do not understand why you make these two sums. Especially as they are equal in value.